vmproxy/pkg/proxy/proxy_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2022 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package proxy
+
+import (
+	"encoding/base64"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	headerName = "x-test-header"
+	targetURL  = "http://127.0.0.1"
+)
+
+func TestProxy(t *testing.T) {
+	t.Parallel()
+
+	setup := func(t *testing.T, filters []string) http.HandlerFunc {
+		t.Helper()
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if filters != nil {
+				require.Equal(t, r.URL.RawQuery, url.Values{"extra_filters": filters}.Encode())
+			}
+		}))
+		t.Cleanup(func() {
+			server.Close()
+		})
+
+		testURL, err := url.Parse(server.URL)
+		require.NoError(t, err)
+
+		handler := getHandler(Config{
+			HeaderName: headerName,
+			TargetURL:  testURL,
+		})
+
+		return handler
+	}
+
+	handler := setup(t, nil)
+
+	t.Run("shall proxy request", func(t *testing.T) {
+		t.Parallel()
+
+		rec := httptest.NewRecorder()
+		req := httptest.NewRequest(http.MethodGet, targetURL, nil)
+
+		handler.ServeHTTP(rec, req)
+		require.Equal(t, rec.Result().StatusCode, http.StatusOK)
+	})
+
+	t.Run("shall properly handle filters", func(t *testing.T) {
+		t.Parallel()
+
+		type testParams struct {
+			expectedFilters []string
+			expectedStatus  int
+			headerContent   string
+			name            string
+		}
+
+		testCases := []testParams{
```

Comment:
btw, the real cases will be in format `[{<key>="<value>"}]`
```
[{region="west"}]
[{region="west", project="app"}]

[{region="west", project="app"}, {region="east", project="shop"}]
```

Reply:
That's fine. These are unit tests and vmproxy does not enforce any format.